perses/tests/test_topology_proposal.py

```diff
@@ -72,6 +74,57 @@ def test_small_molecule_proposals():
         assert smiles == proposal.new_chemical_state_key
         proposal = new_proposal
 
+
+def test_small_molecule_constraint_repair_mapping():
+    """
+    Tests that a constrained edge is demapped after transformed to an unconstrained one.
+
+    This is done by checking that the new_to_old atom map doesn't have the atom that otherwise would be included.
+
+    For the ligands in the test this atom is the ONLY Fluorine atom involved in the transformation.
+    """
+    # TODO: we could try testing a pyridine to a fluoropyridine transformation with using the given geometries
+    from perses.utils.openeye import createOEMolFromSDF
+    # Get sdfs paths
+    sdf_path = resource_filename(
+        "perses",
+        os.path.join("data", "constrained-to-unconstrained", "ligands_constraint_test.sdf"),
+    )
+    # Using troublesome molecules in SDF data
+    ligand_a = createOEMolFromSDF(
+        sdf_path,
+        index=0)
+    ligand_b = createOEMolFromSDF(
+        sdf_path,
+        index=1)
+    # Set title to 'MOL'
+    ligand_a.SetTitle('MOL')
+    ligand_b.SetTitle('MOL')
+    # Create system
+    molecules = [ligand_a, ligand_b]
+    off_molecules = [Molecule.from_openeye(molecule) for molecule in molecules]
+
+    system_generator = SystemGenerator(forcefields=forcefield_files, barostat=barostat,
+                                       forcefield_kwargs=forcefield_kwargs,
+                                       nonperiodic_forcefield_kwargs=nonperiodic_forcefield_kwargs,
+                                       small_molecule_forcefield=small_molecule_forcefield, molecules=off_molecules,
+                                       cache=None)
+    # Propose topology
+    proposal_engine = SmallMoleculeSetProposalEngine(molecules, system_generator)
+    proposal_engine.use_given_geometries = True  # Force to use the given geometries in the propose
+    proposal_engine.given_geometries_tolerance = 0.4 * unit.angstrom
+    initial_system, initial_positions, initial_topology = OEMol_to_omm_ff(molecules[0], system_generator)
```

Comment:
You can use `off_molecules[0].to_openeye()` instead of carrying around the OpenEye `OEMol` here:
```python
    initial_system, initial_positions, initial_topology = OEMol_to_omm_ff(off_molecules[0].to_openeye(), system_generator)
```

Reply:
I don't really understand why, but when I try that I get the following traceback https://gist.github.com/ijpulidos/c192994384f61a1843111105db332cfa This was the reason that I decided to use OE molecules, because I did want this to use off molecules in the first place.

Reply:
My guess is that there must be something getting lost when converting from openff's `Molecule` to `OEMol`

Reply:
Yikes! That sounds like something we should chase down separately. I'll look into this.